drivers/cleanup.py

```diff
@@ -940,7 +940,7 @@ def _tagChildrenForRelink(self):
         if len(self.children) == 0:
             retries = 0
             try:
-                while retries < 10:
+                while retries < 15:
```

Comment:
The use of DB_VDI_ACTIVATING below is confusing, because it is only ever read when doing a `git grep`. blktap2.py has an `activating` key, but doesn't use this constant, can the constants be shared between the 2 files? Same for relinking

Reply:
The code below uses 2 separate keys to implement a lock, I can see why it is done this way (due to how setConfig is implemented), but it might be better to use a single 'lock' field and instead of remove+add, just use 'add' and catch the exception saying Duplicate that the field is already set (that means someone else is holding a lock). And you'd only ever do a 'removeFromConfigVDI' if you are the one currently holding a lock. That means we wouldn't need the get1 + set2 + get1 + del2 logic at all. For additional safety the value should be globally unique (e.g. host UUID + PID): XAPI only responds with Duplicate if the value is different.

Reply:
Refactoring these constants is a fairly sizeable job as they currently exist only in the VDI subclass in the garbage collection code. But yes, that is definitely something that would improve the readability of the code.

The other point about using a single lock field would need to be looked at carefully, how good are the atomicy guarantees for setConfig? (Maybe that doesn't change anything significantly though)